File: 1-Iniciante/c/1021_notas-e-moedas.c
#include <stdio.h>

int main ()
{
    int notas[6] = {0, 0, 0, 0, 0, 0}, moedas[6] = {0, 0, 0, 0, 0, 0};
    int valor, valor2;
    double informado, valor2_0;

    scanf("%lf", &informado);

    valor = informado;
    valor2 = informado;
    notas[0] = valor/100;
    valor = valor%100;
    notas[1] = valor/50;
    valor = valor%50;
    notas[2] = valor/20;           // notas
    valor = valor%20;
    notas[3] = valor/10;
    valor = valor%10;
    notas[4] = valor/5;
    valor = valor%5;
    notas[5] = valor/2;
    moedas[0] = valor%2;

    valor2_0 = informado*100; // multiplica o que sobrou pra virar um inteiro
    valor2 = (int)valor2_0; // transforma em inteiro

    valor2 = valor2%100;
    moedas[1] = valor2/50;
    valor2 = valor2%50;
    moedas[2] = valor2/25;           // moedas
    valor2 = valor2%25;
    moedas[3] = valor2/10;
    valor2 = valor2%10;
    moedas[4] = valor2/5;
    moedas[5] = valor2%5;

    printf("NOTAS:\n");
    printf("%d nota(s) de R$ 100.00\n", notas[0]);
    printf("%d nota(s) de R$ 50.00\n", notas[1]);
    printf("%d nota(s) de R$ 20.00\n", notas[2]);
    printf("%d nota(s) de R$ 10.00\n", notas[3]);
    printf("%d nota(s) de R$ 5.00\n", notas[4]);
    printf("%d nota(s) de R$ 2.00\n", notas[5]);
    printf("MOEDAS:\n");
    printf("%d moeda(s) de R$ 1.00\n", moedas[0]);
    printf("%d moeda(s) de R$ 0.50\n", moedas[1]);
    printf("%d moeda(s) de R$ 0.25\n", moedas[2]);
    printf("%d moeda(s) de R$ 0.10\n", moedas[3]);
    printf("%d moeda(s) de R$ 0.05\n", moedas[4]);
    printf("%d moeda(s) de R$ 0.01\n", moedas[5]);

   return 0; 
}

File: 1-Iniciante/python/1021_notas-e-moedas.py
informado = input()

if ',' in informado: 
    informado = informado.replace(',', '.')
informado = float(informado)

notas = {'100': 0, '50': 0, '20': 0, '10': 0, '5': 0, '2': 0}
moedas = {'1': 0, '0.50': 0, '0.25': 0, '0.10': 0, '0.05': 0, '0.01': 0}	

while informado > 0.00:
    informado = round(informado, 2)
    if informado >= 100.00:
        notas['100'] += 1
        informado -= 100.00
    elif informado >= 50.00:
        notas['50'] += 1
        informado -= 50.00
    elif informado >= 20.00:
        notas['20'] += 1
        informado -= 20.00
    elif informado >= 10.00:
        notas['10'] += 1
        informado -= 10.00
    elif informado >= 5.00:
        notas['5'] += 1
        informado -= 5.00
    elif informado >= 2.00:
        notas['2'] += 1
        informado -= 2.00
    elif informado >= 1.00:
        moedas['1'] += 1
        informado -= 1.00
    elif informado >= 0.50:
        moedas['0.50'] += 1
        informado -= 0.50
    elif informado >= 0.25:
        moedas['0.25'] += 1
        informado -= 0.25
    elif informado >= 0.10:
        moedas['0.10'] += 1
        informado -= 0.10
    elif informado >= 0.05:
        moedas['0.05'] += 1
        informado -= 0.05
    else:
        moedas['0.01'] += 1
        informado -= 0.01
    
print(f'''NOTAS:
{notas['100']} nota(s) de R$ 100.00
{notas['50']} nota(s) de R$ 50.00
{notas['20']} nota(s) de R$ 20.00
{notas['10']} nota(s) de R$ 10.00
{notas['5']} nota(s) de R$ 5.00
{notas['2']} nota(s) de R$ 2.00
MOEDAS:
{moedas['1']} moeda(s) de R$ 1.00
{moedas['0.50']} moeda(s) de R$ 0.50
{moedas['0.25']} moeda(s) de R$ 0.25
{moedas['0.10']} moeda(s) de R$ 0.10
{moedas['0.05']} moeda(s) de R$ 0.05
{moedas['0.01']} moeda(s) de R$ 0.01''')

